File: android_java/app/src/main/java/com/fjun/androidjavaweatherlabapp/MyApplication.java
package com.fjun.androidjavaweatherlabapp;

import android.app.Activity;
import android.app.Application;

import com.fjun.androidjavaweatherlabapp.di.DaggerAppComponent;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;

public class MyApplication extends Application implements HasActivityInjector {

    @Inject
    DispatchingAndroidInjector<Activity> activityDispatchingAndroidInjector;

    @Override
    public void onCreate() {
        super.onCreate();
        DaggerAppComponent.builder()
                .application(this)
                .build().inject(this);
    }


    @Override
    public AndroidInjector<Activity> activityInjector() {
        return activityDispatchingAndroidInjector;
    }
}

File: android_kotlin/app/src/main/java/com/fjun/android_kotlin/MyApplication.kt
package com.fjun.android_kotlin

import android.app.Application
import org.koin.android.ext.android.startKoin

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        // Start Koin
        startKoin(this, listOf(myModule))
    }
}